crates/sui-sdk/src/apis.rs

```diff
@@ -290,19 +290,28 @@ impl CoinReadApi {
         coin_type: Option<String>,
     ) -> impl Stream<Item = Coin> + '_ {
         stream::unfold(
-            (vec![], None, true, coin_type),
-            move |(mut data, cursor, first, coin_type)| async move {
+            (
+                vec![],
+                /* cursor */ None,
+                /* has_next_page */ true,
+                coin_type,
+            ),
+            move |(mut data, cursor, has_next_page, coin_type)| async move {
                 if let Some(item) = data.pop() {
-                    Some((item, (data, cursor, false, coin_type)))
-                } else if (cursor.is_none() && first) || cursor.is_some() {
+                    Some((item, (data, cursor, /* has_next_page */ true, coin_type)))
```

Comment:
If data.pop, we always set has_next_page to true?

Reply:
yeah, the iteration will stop only when there is no more items on the current page and no more pages;
the reason I did not want to use the "has_next_page" was that, after last request, there is a possibility that new page gets generated, so comparing with saving an extra read, it's prob better to have more accurate has_next_page.